src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,35 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> map = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (map.containsKey(userId)) {
+            return getUser(userId,executor);
+
+        }
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        map.put(userId,new UserData(userId,"some details"));
+        return getUser(userId,executor);
+    }
```

Comment:
The current implementation of 'processRequest' is blocking due to the use of 'Thread.sleep(1000)'. This defeats the purpose of using asynchronous processing. Instead, consider using a delay mechanism that is compatible with CompletableFuture, such as 'CompletableFuture.delayedExecutor' with 'CompletableFuture.supplyAsync', to simulate any necessary delays without blocking the thread.